.ci/bump-golang.yml

```diff
@@ -0,0 +1,117 @@
+---
+name: Bump golang-version to latest version
+
+actions:
+  default:
+    title: '[updatecli] Bump Golang version to {{ source "latestGoVersion" }}'
+    kind: github/pullrequest
+    spec:
+      labels:
+        - backport-skip
+        - dependencies
+    scmid: default
+
+scms:
+  default:
+    kind: github
+    spec:
+      user: '{{ requiredEnv "GIT_USER" }}'
+      email: '{{ requiredEnv "GIT_EMAIL" }}'
+      owner: elastic
+      repository: apm-server
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GIT_USER" }}'
+      branch: main
+
+sources:
+  minor:
+    name: Get minor version in .go-version
+    kind: shell
+    transformers:
+      - findsubmatch:
+          pattern: '^\d+.(\d+).\d+$'
+          captureindex: 1
+    spec:
+      command: cat .go-version
+
+  latestGoVersion:
+    name: Get Latest Go Release
+    kind: githubrelease
+    dependson:
+      - minor
+    transformers:
+      - trimprefix: go
+    spec:
+      owner: golang
+      repository: go
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GIT_USER" }}'
+      versionfilter:
+        kind: regex
+        pattern: go1\.{{ source "minor" }}\.\d*$
+
+  gomod:
+    dependson:
+      - latestGoVersion
+    name: Get version in go.mod format
+    kind: shell
+    transformers:
+      - findsubmatch:
+          pattern: '^(\d+.\d+).\d*'
+          captureindex: 1
+    spec:
+      command: echo {{ source "latestGoVersion" }}
+
+conditions:
+  dockerTag:
+    name: Is docker image golang:{{ source "latestGoVersion" }} published
+    kind: dockerimage
+    spec:
+      image: golang
+      tag: '{{ source "latestGoVersion" }}'
+    sourceid: latestGoVersion
+
+  goDefaultVersion-check:
+    name: Check if defined golang version differs
+    kind: shell
+    sourceid: latestGoVersion
+    spec:
+      command: 'grep -v -q {{ source "latestGoVersion" }} .go-version #'
+
+targets:
+  update-go-version:
+    name: "Update .go-version"
+    sourceid: latestGoVersion
+    scmid: default
+    kind: file
+    spec:
+      content: '{{ source "latestGoVersion" }}'
+      file: .go-version
+      matchpattern: '\d+.\d+.\d+'
+  update-version-asciidoc:
+    name: "Update version.asciidoc"
+    sourceid: latestGoVersion
+    scmid: default
+    kind: file
+    spec:
+      file: docs/version.asciidoc
+      matchpattern: '(:go-version:) \d+.\d+.\d+'
+      replacepattern: '$1 {{ source "latestGoVersion" }}'
+  update-dockerfile:
+    name: "Update Dockerfile"
+    sourceid: latestGoVersion
+    scmid: default
+    kind: file
+    spec:
+      file: packaging/docker/Dockerfile
+      matchpattern: '(FROM golang):\d+.\d+.\d+'
+      replacepattern: '$1:{{ source "latestGoVersion" }}'
+  update-gomod:
```

Comment:
I think this is only targeting the main `go.mod` but we also have other modules that need to be synced:

```
$ rg --files -g go.mod
tools/go.mod
internal/glog/go.mod
go.mod
cmd/intake-receiver/go.mod
systemtest/go.mod
```

Reply:
https://github.com/elastic/apm-server/pull/10243/commits/ef66634e6ad0bf4669b4fe4f598edbf71874f208